openssl-dynamic/src/main/java/org/apache/tomcat/jni/SSL.java

```diff
@@ -551,10 +551,27 @@ public static long makeNetworkBIO(long ssl) {
      * @return pointer to the Network BIO (BIO *)
      */
     public static long makeNetworkBIO(long ssl, int maxBioSize) {
-        return makeNetworkBIO0(ssl, maxBioSize);
+        return makeNetworkBIO0(ssl, maxBioSize, maxBioSize);
     }
 
-    private static native long makeNetworkBIO0(long ssl, int maxBioSize);
+    /**
+     * Wire up internal and network BIOs for the given SSL instance.
+     *
+     * <b>Warning: you must explicitly free this resource by calling freeBIO</b>
+     *
+     * While the SSL's internal/application data BIO will be freed when freeSSL is called on
+     * the provided SSL instance, you must call freeBIO on the returned network BIO.
+     *
+     * @param ssl the SSL instance (SSL *)
+     * @param maxInternalBIOSize The maximum size of the BIO. Pass 0 to use the default max size.
```

Comment:
clarify that this BIO is on the application side. also do the same for the network BIO. Add a link to https://www.openssl.org/docs/man1.0.1/crypto/BIO_s_bio.html (see the example section)